packages/terra-site/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by <ThemeProvider />
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `terra-OSC-theme`
+   */
+  themeName: PropTypes.string.isRequired,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => {
+  return <div className={themeName}>{children}</div>;
```

Comment:
Was discussing with @mhemesath and @zhongyn , updating the theme name would effectively re-render the children which could be the whole site, where really I think it just needs to add the new className, declaring the new variable, and forcing a CSS update.
Could we wrap the children in another component that implements ShouldComponentUpdate, or extends React.PureComponent to avoid re-rendering the whole page on theme change?